### PR TITLE
Update EGit/Wikitext references to JavaSE-{1.8,11| to JavaSE-17 #393

### PR DESCRIPTION
**Pull request by gnl42**
Task-Url: https://github.com/eclipse-mylyn/org.eclipse.mylyn/issues/393

### CODE REVIEW

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant.tests/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -5,7 +5,7 @@ Bundle-Version: 4.2.0.qualifier
 Bundle-Name: Mylyn WikiText Ant Tasks Tests
 Bundle-Vendor: Eclipse Mylyn
 Fragment-Host: org.eclipse.mylyn.wikitext.ant
-Bundle-RequiredExecutionEnvironment: JavaSE-11
+Bundle-RequiredExecutionEnvironment: JavaSE-17
 Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=11))"
```
Comment (akurtakov) [inline mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant.tests/META-INF/MANIFEST.MF:9]: You have to update the osgi.ee require capability to Java 17 too.

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -4,7 +4,7 @@ Bundle-SymbolicName: org.eclipse.mylyn.wikitext.ant
 Bundle-Version: 4.2.0.qualifier
 Bundle-Name: Mylyn WikiText Ant Tasks
 Bundle-Vendor: Eclipse Mylyn
-Bundle-RequiredExecutionEnvironment: JavaSE-11
+Bundle-RequiredExecutionEnvironment: JavaSE-17
 Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=11))"
```
Comment (akurtakov) [inline mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant/META-INF/MANIFEST.MF:8]: Update osgi.ee too.

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc.tests/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -5,7 +5,7 @@ Bundle-Version: 4.2.0.qualifier
 Bundle-Name: Mylyn WikiText AsciiDoc Tests
 Bundle-Vendor: Eclipse Mylyn
 Fragment-Host: org.eclipse.mylyn.wikitext.asciidoc
-Bundle-RequiredExecutionEnvironment: JavaSE-11
+Bundle-RequiredExecutionEnvironment: JavaSE-17
 Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=11))"
```
Comment (akurtakov) [inline mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc.tests/META-INF/MANIFEST.MF:9]: And here.

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -4,7 +4,7 @@ Bundle-SymbolicName: org.eclipse.mylyn.wikitext.asciidoc
 Bundle-Version: 4.2.0.qualifier
 Bundle-Name: Mylyn WikiText AsciiDoc
 Bundle-Vendor: Eclipse Mylyn
-Bundle-RequiredExecutionEnvironment: JavaSE-11
+Bundle-RequiredExecutionEnvironment: JavaSE-17
 Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=11))"
```
Comment (akurtakov) [inline mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc/META-INF/MANIFEST.MF:8]: And here.

**Review by akurtakov — COMMENTED** [review]
Every osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=11))" has to become osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=17))" . I don't know what happens when BREE and osgi.ee capability conflict but it's a case we would rather not try. 
BREE is deprecated and osgi.ee is the correct replacement so in theory BREE shouldn't be needed at all but I'm not sure whether all of PDE accepts that. @HannessWell any comment?

**Review by merks — APPROVED** [review]
Other than the inconsistencies mentioned by @akurtakov  it looks fine.

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant.tests/META-INF/MANIFEST.MF (hunk 1 above)
Comment (gnl42) [reply mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant.tests/META-INF/MANIFEST.MF]: Fixed

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant/META-INF/MANIFEST.MF (hunk 2 above)
Comment (gnl42) [reply mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.ant/META-INF/MANIFEST.MF]: Fixed

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc.tests/META-INF/MANIFEST.MF (hunk 3 above)
Comment (gnl42) [reply mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc.tests/META-INF/MANIFEST.MF]: Fixed

File: mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc/META-INF/MANIFEST.MF (hunk 4 above)
Comment (gnl42) [reply mylyn.docs/wikitext/core/org.eclipse.mylyn.wikitext.asciidoc/META-INF/MANIFEST.MF]: Fixed

**Comment by gnl42** [comment]
A fair number of the MANIFEST.MF files doesn't have Require-Capability:

**Review by akurtakov — APPROVED** [review]

**Comment by akurtakov** [comment]
> A fair number of the MANIFEST.MF files doesn't have Require-Capability:

This is fine.

**Pull request merged**